package/MDAnalysis/analysis/lineardensity.py

```diff
@@ -146,13 +150,19 @@ def _prepare(self):
         group = getattr(self._ags[0], self.grouping)
 
         # Get masses and charges for the selection
-        try:  # in case it's not an atom
-            self.masses = np.array([elem.total_mass() for elem in group])
-            self.charges = np.array([elem.total_charge() for elem in group])
-        except AttributeError:  # much much faster for atoms
+        if self.grouping == "atoms":
```

Comment:
rather than using 3 `if` it might be worth making this `if/elif/else`? Possibly throwing an error if you have an unknown grouping?